/*
 * Copyright (c) 2022 Huawei Technologies Co., Ltd
 *
 * This software is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *
 * http://license.coscl.org.cn/MulanPSL2
 *
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

syntax = "proto3";

package common;

option go_package = "grpc/pb/common;common";

message Arg {
  enum ArgType {
    VALUE      = 0;
    OBJECT_REF = 1;
  }
  ArgType type  = 1;
  bytes   value = 2;
  repeated string nested_refs = 3;
}

enum ErrorCode {
  ERR_NONE                        = 0;
  ERR_PARAM_INVALID               = 1001;
  ERR_RESOURCE_NOT_ENOUGH         = 1002;
  ERR_INSTANCE_NOT_FOUND          = 1003;
  ERR_INSTANCE_DUPLICATED         = 1004;
  ERR_INVOKE_RATE_LIMITED         = 1005;
  ERR_USER_CODE_LOAD              = 2001;
  ERR_USER_FUNCTION_EXCEPTION     = 2002;
  ERR_REQUEST_BETWEEN_RUNTIME_BUS = 3001;
  ERR_INNER_COMMUNICATION         = 3002;
  ERR_INNER_SYSTEM_ERROR          = 3003;
}
